czkawka_core/src/duplicate.rs

```diff
@@ -1297,6 +1298,17 @@ fn delete_files(vector: &[FileEntry], delete_method: &DeleteMethod, warnings: &m
     (gained_space, removed_files, failed_to_remove_files)
 }
 
+fn make_hard_link(src: &PathBuf, dst: &PathBuf) -> Result<()> {
+    let dst_dir = dst.parent().ok_or_else(|| Error::new(ErrorKind::Other, "No parent"))?;
+    let temp = tempfile::Builder::new().tempfile_in(dst_dir)?;
+    fs::rename(dst, temp.path())?;
```

Comment:
On Windows it's not possible to rename a file if the "target" file exists. Because the `tempfile_in` function creates a file (not just a name/path), the rename will fail with a permission denied error.

```
Error: Os { code: 5, kind: PermissionDenied, message: "Odmowa dostępu." }
```

Reply:
I've no windows machine. The documentation says https://doc.rust-lang.org/std/fs/fn.rename.html:
> This function currently corresponds to the rename function on Unix and the MoveFileEx function with the MOVEFILE_REPLACE_EXISTING flag on Windows.

I assume it's actually a different root cause, e.g. that the files are on different file systems. You could try to do the same file operations manually and see where it fails.